rundeckapp/grails-app/controllers/rundeck/controllers/MenuController.groovy

```diff
@@ -577,11 +593,32 @@ class MenuController extends ControllerBase implements ApplicationContextAware{
                 formatted
         )
     }
+
+    private def getConfiguredMaxPerPage(int defaultMax) {
+        if(guiPaginationEnabled) {
+            return grailsApplication.config.rundeck.gui.paginatejobs.max.per.page.isEmpty() ? defaultMax : grailsApplication.config.rundeck.gui.paginatejobs.max.per.page.toInteger()
+        }
+        return null
+    }
+
+    private boolean getGuiPaginationEnabled() {
+        grailsApplication.config.rundeck.gui.paginatejobs.enabled in ["true",true]
+    }
+
+    private boolean getApiPaginationEnabled() {
+        if(grailsApplication.config.rundeck.api.paginatejobs.enabled != null)
+            return grailsApplication.config.rundeck.api.paginatejobs.enabled in ["false",false]
+        
+        return true
```

Comment:
should use the `configurationService` instead of grailsApplication.config

Reply:
I had to add the classes Max and Per to access paginatejobs.max.per.page property from ConfigurationService without affecting compatibility...